backends/platform/sdl/amigaos/amigaos-main.cpp

```diff
@@ -24,13 +24,44 @@
 
 #if defined(__amigaos4__)
 
+#include "backends/fs/amigaos4/amigaos4-fs.h"
 #include "backends/platform/sdl/amigaos/amigaos.h"
 #include "backends/plugins/sdl/sdl-provider.h"
 #include "base/main.h"
 
 int main(int argc, char *argv[]) {
 
-	// Set up a stack cookie to avoid crashes due to too few stack set by users 
+	// The following will gather the application name and add the install path
+	// to a variable in AmigaOS4's ENV(ARC) system. It will be placed in AppPaths
+	// so that ScummVM can become AmiUpdate aware
+	const char *const appname = "ScummVM";
+
+	BPTR lock;
+	APTR oldwin;
+
+		// Obtain the lock to the home directory
+		if(( lock = IDOS->GetProgramDir() )) {
```

Comment:
If i remove the second set of parentheses i get a compiler warning:

    C++      backends/platform/sdl/amigaos/amigaos-main.o
backends/platform/sdl/amigaos/amigaos-main.cpp: In function 'int main(int, char**)':
backends/platform/sdl/amigaos/amigaos-main.cpp:43:34: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
  if (lock = IDOS->GetProgramDir()) {
                                   (Points to the last parantheses)    ^